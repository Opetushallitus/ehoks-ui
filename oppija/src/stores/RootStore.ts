// import kebabCase from "lodash.kebabcase"
import { getEnv, Instance, types } from "mobx-state-tree"
import { EnvironmentStore } from "stores/EnvironmentStore"
import { HOKSStore } from "stores/HOKSStore"
// TODO: remove KoulutuksenJarjestajaStore when LuoHOKS route is removed
import { KoulutuksenJarjestajaStore } from "stores/KoulutuksenJarjestajaStore"
import { OppilasStore } from "stores/OppilasStore"
import { SessionStore } from "stores/SessionStore"
import { TranslationStore } from "stores/TranslationStore"
import { TyopaikanToimijaStore } from "stores/TyopaikanToimijaStore"

export interface InjectedStores {
  store: IRootStore
}

const RootStoreModel = {
  environment: types.optional(EnvironmentStore, {}),
  hoks: types.optional(HOKSStore, {}),
  isLoading: false,
  koulutuksenJarjestaja: types.optional(KoulutuksenJarjestajaStore, {}),
  oppilas: types.optional(OppilasStore, {}),
  session: types.optional(SessionStore, {}),
  translations: types.optional(TranslationStore, {}),
  tyopaikanToimija: types.optional(TyopaikanToimijaStore, {})
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => {
    return {
      get errors() {
        return getEnv(self).errors
      }
    }
  })

export interface IRootStore extends Instance<typeof RootStore> {}
