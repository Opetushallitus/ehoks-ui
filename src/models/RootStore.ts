import camelCase from "lodash.camelcase"
// import kebabCase from "lodash.kebabcase"
import mapObj from "map-obj"
import { flow, getEnv, types } from "mobx-state-tree"
import { EducationProviderStore } from "models/EducationProviderStore"
import { StudentStore } from "models/StudentStore"
import { WorkplaceProviderStore } from "models/WorkPlaceProviderStore"
import { LearningPeriod } from "./LearningPeriod"

const RootStoreModel = {
  activeLocale: types.optional(
    types.union(types.literal("fi"), types.literal("sv")),
    "fi"
  ),
  education: types.optional(EducationProviderStore, { info: {} }),
  isLoading: false,
  learningPeriods: types.optional(types.array(LearningPeriod), []),
  student: types.optional(StudentStore, { info: {} }),
  work: types.optional(WorkplaceProviderStore, { info: {} })
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => ({
    // views defined here should be used from other stores using: getRoot(self)
    get fetch() {
      return getEnv(self).fetch
    }
  }))
  .actions(self => {
    const fetchSingle = flow(function*(url: string) {
      const response = yield self.fetch(url)
      const model = response.data && response.data[0] ? response.data[0] : {}
      // camelCase kebab-cased object keys recursively so we can use dot syntax for accessing values
      return mapObj(
        model,
        (key: string, value: any) => [camelCase(key), value],
        {
          deep: true
        }
      )
    })

    const fetchCollection = flow(function*(url: string) {
      const response = yield self.fetch(url)
      console.log("fetchCollection", response)
      const model = response.data ? response.data : []
      return model
    })

    return { fetchSingle, fetchCollection }
  })
