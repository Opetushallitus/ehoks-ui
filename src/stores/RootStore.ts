// import kebabCase from "lodash.kebabcase"
import { getEnv, Instance, types } from "mobx-state-tree"
import { EducationProviderStore } from "stores/EducationProviderStore"
import { EnvironmentStore } from "stores/EnvironmentStore"
import { OppilasStore } from "stores/OppilasStore"
import { SessionStore } from "stores/SessionStore"
import { TranslationStore } from "stores/TranslationStore"
import { TyopaikanToimijaStore } from "stores/TyopaikanToimijaStore"

export interface ApiResponse<T> {
  meta: {
    [name: string]: any
  }
  data: T
}

export interface InjectedStores {
  store: IRootStore
}

const RootStoreModel = {
  education: types.optional(EducationProviderStore, { info: {} }),
  environment: types.optional(EnvironmentStore, {}),
  isLoading: false,
  oppilas: types.optional(OppilasStore, {}),
  session: types.optional(SessionStore, {}),
  translations: types.optional(TranslationStore, {}),
  tyopaikanToimija: types.optional(TyopaikanToimijaStore, {})
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => {
    return {
      get errors() {
        return getEnv(self).errors
      }
    }
  })

export interface IRootStore extends Instance<typeof RootStore> {}
