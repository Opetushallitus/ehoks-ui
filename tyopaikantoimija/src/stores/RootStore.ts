import { getEnv, Instance, types } from "mobx-state-tree"
import { EnvironmentStore } from "stores/EnvironmentStore"
import { IErrorStore } from "stores/ErrorStore"
import { SessionStore } from "stores/SessionStore"
import { TranslationStore } from "stores/TranslationStore"

export interface InjectedStores {
  store: IRootStore
}

const RootStoreModel = {
  environment: types.optional(EnvironmentStore, {}),
  isLoading: false,
  session: types.optional(SessionStore, {}),
  translations: types.optional(TranslationStore, {})
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => ({
    get errors(): IErrorStore {
      return getEnv(self).errors
    }
  }))

export type IRootStore = Instance<typeof RootStore>
