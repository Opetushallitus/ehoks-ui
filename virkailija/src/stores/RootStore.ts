import { getEnv, Instance, types } from "mobx-state-tree"
import { EnvironmentStore } from "./EnvironmentStore"
import { SessionStore } from "./SessionStore"
import { TranslationStore } from "stores/TranslationStore"
import { KoulutuksenJarjestajaStore } from "./KoulutuksenJarjestajaStore"
import { NotificationStore } from "./NotificationStore"
import { RaportitStore } from "./RaportitStore"

export interface InjectedStores {
  store: IRootStore
}

const RootStoreModel = {
  environment: types.optional(EnvironmentStore, {}),
  koulutuksenJarjestaja: types.optional(KoulutuksenJarjestajaStore, {}),
  session: types.optional(SessionStore, {}),
  translations: types.optional(TranslationStore, {}),
  notifications: types.optional(NotificationStore, {}),
  raportit: types.optional(RaportitStore, {})
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => ({
    get errors() {
      return getEnv(self).errors
    }
  }))

export type IRootStore = Instance<typeof RootStore>
