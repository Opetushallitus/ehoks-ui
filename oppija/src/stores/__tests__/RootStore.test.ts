import { getSnapshot } from "mobx-state-tree"
import { RootStore } from "../RootStore"

describe("RootStore", () => {
  test("constructor should produce valid tree even with empty input object", () => {
    const store = RootStore.create({}, {})
    expect(getSnapshot(store)).toMatchSnapshot()
  })
})
