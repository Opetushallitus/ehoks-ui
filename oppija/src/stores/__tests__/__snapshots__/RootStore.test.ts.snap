// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootStore constructor should produce valid tree even with empty input object 1`] = `
Object {
  "environment": Object {
    "casOppijaLoginUrlFi": "",
    "casOppijaLoginUrlSv": "",
    "casOppijaLogoutUrlFi": "",
    "casOppijaLogoutUrlSv": "",
    "eperusteetPerusteUrl": "",
    "error": "",
    "isLoading": false,
    "opintopolkuLoginUrlFi": "",
    "opintopolkuLoginUrlSv": "",
    "opintopolkuLogoutUrlFi": "",
    "opintopolkuLogoutUrlSv": "",
    "virkailijaLoginUrl": "",
  },
  "hoks": Object {
    "isLoading": false,
    "suunnitelmat": Array [],
  },
  "isLoading": false,
  "notifications": Object {
    "notifications": Array [],
    "studentFeedbackLinks": Array [],
  },
  "oppilas": Object {
    "isLoading": false,
    "perusteet": Array [],
  },
  "session": Object {
    "error": "",
    "isLoading": false,
    "selectedOrganisationOid": "",
    "settings": Object {
      "hiddenNotifications": Object {
        "notifications": Array [],
      },
      "introDialog": Object {
        "userAcknowledgedIntroDialog": false,
      },
    },
    "user": undefined,
    "userDidLogout": false,
  },
  "shareLinks": Object {},
  "translations": Object {
    "activeLocale": "fi",
    "isLoading": false,
    "translations": Array [],
  },
}
`;
